reactive/coredns.py

```diff
@@ -45,5 +88,40 @@ def start_charm():
 
 @when('layer.docker-resource.coredns-image.changed')
 def update_image():
-    # handle a new image resource becoming available
     clear_flag('charm.coredns.started')
+
+
+@when('charm.coredns.started', 'cluster-dns.domain-ready')
+def send_ip():
+    """
+    Send CoreDNS IP to kuberentes-worker
+    """
+    try:
+        cluster_dns = endpoint_from_flag('cluster-dns.domain-ready')
+        if cluster_dns:
+            service_ip = get_service_ip('coredns')
```

Comment:
Do we have a plan for upgrading existing CDK clusters from the cdk-addons based coredns to the charm-based coredns?

I noticed here that the service name is `coredns`, not `kube-dns`, which means that migrating from cdk-addons will result in a new DNS service name, meaning a new DNS service IP, meaning all pods will need to be recreated to pick up the new DNS IP. That's a big disruption to user workloads and I don't think it's something we could reasonably ship.

Reply:
True, does the fact that the service will be deployed outside of the `kube-system`  namespace also affect this?

Reply:
Oh yeah, it does. If the service becomes anything besides `kube-dns` in the `kube-system` namespace, then upgrading to the new service will be disruptive.

Reply:
So unless we are able to deploy kubernetes charms into `kube-system` this will always be disruptive.